### ISSUE TITLE
UI:run-local unable to load meshery ui

### ISSUE THREAD

**Issue by nitishvu** [layer5io/meshery#958]
**Description**
After executing "make run-local"  some times unable to load UI
getting below error in the terminal. no error in the browser console

<pre><font color="#06989A">INFO</font>[0000] Using &apos;/home/nitish/.meshery/config&apos; to store user data 
<font color="#06989A">INFO</font>[0000] Using &apos;/home/nitish/.kube&apos; as the folder to look for kubeconfig file 
<font color="#06989A">INFO</font>[0000] Log level: debug                             
<font color="#06989A">INFO</font>[0000] Starting Server listening on :9081           
<font color="#CC0000">ERRO</font>[0041] error converting JSON response to map : json: cannot unmarshal string into Go value of type []map[string]string 
<font color="#CC0000">ERRO</font>[0041] error fetching JWK corresponding to token : json: cannot unmarshal string into Go value of type []map[string]string 
<font color="#06989A">INFO</font>[0041] Token validation error : json: cannot unmarshal string into Go value of type []map[string]string 
<font color="#CC0000">ERRO</font>[0070] token extraction error : unexpected end of JSON input 
<font color="#CC0000">FATA</font>[0070] error decoding token : unexpected end of JSON input 
<font color="#4E9A06">
</pre>

**Expected Behavior**
Meshery UI should load properly

**Screenshots**
cookies in browser
![image](https://user-images.githubusercontent.com/24861433/80981201-459d9980-8e47-11ea-89a5-6f4812fd52c9.png)

**Environment:**
 - OS:  Ubuntu 20.04
 - Browser:  Chrome, Safari
 - Version: code from master
 - Device: desktop

---
[Optional] **To Reproduce**

[Optional] **Additional Context**

**Comment by kanishkarj** [comment]
Hey @nitishvu, when you say unable to load UI, do you mean that even provider UI doesn't show up? The provider UI being the page where you choose between meshery/None before you sign in.

**Comment by nitishvu** [comment]
yes @kanishkarj even provider UI doesn't show up.

**Comment by kanishkarj** [comment]
Oh.. Could you check once if it works on firefox? Because we came across this issue 2 days back. Also, could you verify if your master is up to date with the master on this repo?

**Comment by sb1975** [comment]
I also faced issue in opening Meshery GUI.
either using the Meshery Provide or "None", it gives me below error
esudbat@istio:~$ sudo mesheryctl logs
Starting Meshery logging...
Attaching to meshery_meshery-cpx_1, meshery_meshery_1, meshery_meshery-linkerd_1, meshery_watchtower_1, meshery_meshery-nsm_1, meshery_meshery-consul_1, meshery_meshery-octarine_1, meshery_meshery-istio_1
meshery-consul_1    | time="2020-05-04T16:39:56Z" level=info msg="Serving gRPC on :10002"
..
meshery_1           | time="2020-05-04T16:40:10Z" level=error msg="k8s config is nil"
meshery_1           | time="2020-05-04T16:40:10Z" level=error msg="k8s config is nil"

meshery_1           | time="2020-05-04T17:08:03Z" level=error msg="k8s config is nil"
meshery_1           | time="2020-05-04T17:08:03Z" level=error msg="k8s config is nil"
meshery-istio_1     | WARNING: 2020/05/04 17:41:43 grpc: Server.Serve failed to create ServerTransport: connection error: desc = "transport: http2Server.HandleStreams received bogus greeting from client: \"GET /admin/assets/js/vie\""
meshery-istio_1     | WARNING: 2020/05/04 17:41:43 grpc: Server.Serve failed to create ServerTransport: connection error: desc = "transport: http2Server.HandleStreams received bogus greeting from client: \"GET /admin/assets/js/vie\""

**Comment by leecalcote** [comment]
Gents, the latest Meshery Docker image the image id of `9773ffc8c9a9` fixes this issue. If you run a `mesheryctl start`, this will both update the Meshery container images and run Meshery. Alternatively, you could run `mesheryctl update` and then `mesheryctl start` to pull the latest images and run Meshery.

**Comment by leecalcote** [comment]
There seems to be a dangling issue here when comparing results. If you select two results and compare them, a javascript issue occurs:

```
TypeError: undefined is not an object (evaluating 'this[this.axis.getId(t)].domain')
```
The error points to this function:
```
  getShapeYMin: function(t) {
                            var e = s(this[this.axis.getId(t)].domain(), 1)[0];
                            return !this.isGrouped(t) && e > 0 ? e : 0
                        },
```

@kushthedude @aayushdutt, will you have a look?

**Comment by leecalcote** [comment]
@aayushdutt will you confirm this issue remains? Let's prioritize its fix, if so.

**Comment by aayushdutt** [comment]
Looking into it.

**Comment by aayushdutt** [comment]
@leecalcote I am unable to reproduce the above issue. Maybe it is fixed?
I have tested on no mesh results and it worked perfectly.

**Comment by aayushdutt** [comment]
@kushthedude Could you please confirm if the issue persists?

**Comment by leecalcote** [comment]
@aayushdutt the portion of the issue that I'm able to consistently reproduce is a failure to be able to compare test results (check two or more test result boxes and then click the two-way arrow icon at the top right corner of the results table).

**Comment by aayushdutt** [comment]
@leecalcote  Hmm... This is strange
I did the same and it is working for me. This is what I see:
![image](https://user-images.githubusercontent.com/25280541/81325456-032dc400-90b6-11ea-8a16-0fcf42d68a50.png)
with no errors in the console.

I tested it again with two new tests.
* I created two tests (both with no mesh selected)
* Selected both of them and compared.
* I made sure that the code has the latest changes from upstream.

Could you please show me what you see and/or the configuration you used for the tests?

**Comment by kushthedude** [comment]
Did you run make setup-ui-libs and make build-ui after pulling the latest
changes from upstream?

**Comment by leecalcote** [comment]
No, I'm running the prebuilt release as a user.

**Comment by kushthedude** [comment]
I was asking @aayushdutt ,as I am also facing this issue.

**Comment by aayushdutt** [comment]
I was testing on `make run-ui-dev`, however after doing make `build-ui` and testing on port 9081, I still wasn't able to reproduce the error. Here was what I saw:
![Meshery (1)](https://user-images.githubusercontent.com/25280541/81368964-c641eb80-910e-11ea-9426-a244ea0ad70c.gif)

But after a few tries, I successfully reproduced the error, it is like this: 
![Meshery](https://user-images.githubusercontent.com/25280541/81368759-3b60f100-910e-11ea-990b-e7beb6b29260.gif)

The error is
```js
TypeError: Cannot read property 'domain' of undefined
    at t.getShapeYMin (8326b37b.01d8b36072244d251170.js:1)
    at l (8326b37b.01d8b36072244d251170.js:1)
    at l (82cbf480985f34d90f95ca6942c78235035f9539.59718814718bedff45c0.js:1)
    at SVGPathElement.<anonymous> (8326b37b.01d8b36072244d251170.js:1)
    at SVGPathElement.<anonymous> (82cbf480985f34d90f95ca6942c78235035f9539.59718814718bedff45c0.js:1)
    at G.each (82cbf480985f34d90f95ca6942c78235035f9539.59718814718bedff45c0.js:1)
    at G.attr (82cbf480985f34d90f95ca6942c78235035f9539.59718814718bedff45c0.js:1)
    at t.redrawArea (8326b37b.01d8b36072244d251170.js:1)
    at t.value (8326b37b.01d8b36072244d251170.js:1)
    at t.value (8326b37b.01d8b36072244d251170.js:1)
```

**Comment by kushthedude** [comment]
You have to use 3000 port with make run-ui-dev.
Using 9081 will supress most of the warning/error and also console will not
show anything.

**Comment by aayushdutt** [comment]
@kushthedude Exactly, that's why I was testing on `make run-ui-dev` and on port 3000 earlier.

**Comment by kushthedude** [comment]
Let's open a new issue and move the discussion over there!

**Comment by leecalcote** [comment]
All, just a follow up here - the final bug here is now fixed in #978. Thank you, @kushthedude @aayushdutt
